<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>315</width>
    <height>512</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>315</width>
    <height>512</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Argon2 UI</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="hashTab">
       <attribute name="title">
        <string>Hash</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QVBoxLayout" name="hashTabVerticalLayout">
          <item>
           <layout class="QVBoxLayout" name="passwordVerticalLayout">
            <item>
             <widget class="QLabel" name="passwordLabel">
              <property name="text">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="passwordHorizontalLayout">
              <item>
               <widget class="QLineEdit" name="passwordLineEdit">
                <property name="echoMode">
                 <enum>QLineEdit::Password</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="showPasswordButton">
                <property name="maximumSize">
                 <size>
                  <width>64</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Show</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QWidget" name="hashAlgorithmRadioButtonWidget" native="true">
            <layout class="QHBoxLayout" name="algoHorizontalLayout">
             <item>
              <widget class="QRadioButton" name="argon2idRadioButton">
               <property name="text">
                <string>Argon2id</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">hashAlgorithmButtonGroup</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="argon2iRadioButton">
               <property name="text">
                <string>Argon2i</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">hashAlgorithmButtonGroup</string>
               </attribute>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="argon2dRadioButton">
               <property name="text">
                <string>Argon2d</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">hashAlgorithmButtonGroup</string>
               </attribute>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="hashParamsVerticalLayout">
            <item>
             <widget class="Line" name="hashParamsSeparatorLine">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="timeCostLabel">
              <property name="text">
               <string>Time cost (16 iterations)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="timeCostHorizontalSlider">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
              <property name="value">
               <number>16</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="memoryCostLabel">
              <property name="text">
               <string>Memory cost (32 MiB)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="memoryCostHorizontalSlider">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
              <property name="value">
               <number>32</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="parallelismLabel">
              <property name="text">
               <string>Parallelism (2 threads)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="parallelismHorizontalSlider">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>16</number>
              </property>
              <property name="value">
               <number>2</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="hashLengthLabel">
              <property name="text">
               <string>Hash length (64 B)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="hashLengthHorizontalSlider">
              <property name="minimum">
               <number>16</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
              <property name="value">
               <number>64</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="hashButtonsHorizontalLayout">
            <property name="topMargin">
             <number>4</number>
            </property>
            <item>
             <widget class="QPushButton" name="hashButton">
              <property name="text">
               <string>Hash</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="clearButton">
              <property name="text">
               <string>Clear</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="outputVerticalLayout">
            <item>
             <widget class="QLabel" name="encodedHashLabel">
              <property name="text">
               <string>Output</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="encodedHashTextEdit">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="verifyTab">
       <attribute name="title">
        <string>Verify</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QVBoxLayout" name="inputVerticalLayout">
          <item>
           <widget class="QLabel" name="inputLabel">
            <property name="text">
             <string>Input</string>
            </property>
            <property name="buddy">
             <cstring>hashAlgorithmRadioButtonWidget</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="inputTextEdit">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="inputPasswordFieldVerticalLayout">
          <item>
           <widget class="QLabel" name="inputPasswordLabel">
            <property name="text">
             <string>Password</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="inputPasswordFieldHorizontalLayout">
            <item>
             <widget class="QLineEdit" name="inputPasswordLineEdit">
              <property name="echoMode">
               <enum>QLineEdit::Password</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="showInputPasswordButton">
              <property name="maximumSize">
               <size>
                <width>64</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Show</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>256</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="Line" name="verifyButtonsSeparatorLine">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="verifyButtonsHorizontalLayout">
          <item>
           <widget class="QPushButton" name="verifyButton">
            <property name="text">
             <string>Verify</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearVerificationFieldsButton">
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="aboutTab">
       <attribute name="title">
        <string>About</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QVBoxLayout" name="aboutTabVerticalLayout">
          <item>
           <layout class="QVBoxLayout" name="aboutTabInfoVerticalLayout">
            <item>
             <widget class="QLabel" name="aboutTextLabel">
              <property name="text">
               <string>About Argon2 UI - v1.0.0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="aboutTextEdit">
              <property name="undoRedoEnabled">
               <bool>false</bool>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="aboutTabLegalNoticeVerticalLayout">
            <item>
             <widget class="QLabel" name="aboutTabLegalNoticeLabel">
              <property name="text">
               <string>Qt | Legal notice</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="aboutTabLegalNoticeTextEdit"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>315</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="hashAlgorithmButtonGroup"/>
 </buttongroups>
</ui>
